### ISSUE TITLE
Inline suggestion not visible in controlled mode

### ISSUE THREAD

**Issue by tko** [ericgio/react-bootstrap-typeahead#197]
### Version
1.4.0

### Steps to reproduce
1. type the first character of one of the options

### Expected Behavior
A matching option is shown as dimmed suggestion in the entry.
<img width="224" alt="uncontrolled" src="https://user-images.githubusercontent.com/13787/27080721-6a27ec2e-503d-11e7-94ea-75b04fac2d4e.png">

### Actual Behavior
No suggestion shown, only the typed character and the dropdown.
<img width="222" alt="controlled" src="https://user-images.githubusercontent.com/13787/27080720-6a27d9c8-503d-11e7-8abc-ef16a56dd86a.png">

### Code
```jsx
<Typeahead
    onInputChange={(selected) => this.setState({ selected })}
    options={['one', 'two', 'three']}
    selected={[this.state.selected]}
/>
```

If you change `selected` to `defaultSelected` it works as expected.

**Comment by ericgio** [comment]
Don't put `selected` in an array when passing it in as a prop; it's already an array:

```jsx
<Typeahead
    onInputChange={(selected) => this.setState({selected})}
    options={['one', 'two', 'three']}
    selected={this.state.selected}
/>
```

**Comment by tko** [comment]
I'm using `onInputChange` here, not `onChange` so `selected` should be a string and needs to be wrapped in array when passed back, right?
(My original code was using onChange but this reduced test case seems to behave the same, though I may have missed some subtleties.)

**Comment by ericgio** [comment]
Ah I missed that. Yeah, don't use `onInputChange`. That will fire every time the user types in the input, so it doesn't represent a selection.

**Comment by tko** [comment]
Well, yes, but when used as controlled component isn't that your problem? :)

**Comment by ericgio** [comment]
No. The value passed to `selected` should correspond to one of the options. If you use `onInputChange`, you're potentially passing an arbitrary string (eg: "thr") and the component won't work correctly.

**Comment by tko** [comment]
Hmm, so what's the intended way to handle `onChange` and `onInputChange` events so the component does work correctly in presence of both valid options and arbitrary string? Only `onChange` can be handled as controlled and `onInputChange` _must_ be handled as uncontrolled?

**Comment by ericgio** [comment]
Can you explain what you're trying to do?

`onChange` and `onInputChange` are not analogous and should not be used interchangeably:
- The value passed to `onChange` represents the selected value (string or object), and corresponds to one of the values in `options`.
- The value passed to `onInputChange` represents the text in the input, and could be any arbitrary string.

**Comment by tko** [comment]
I'm trying to set up two Typeahead entries to get the name+email for a person. I have a list of previously known people so I'm feeding it as options to both, the idea being that if you select one of them _both_ of the entries update to match.

When one starts typing instead I'm creating a new option on the fly that is not present in the `options` list and amending that object depending on which entry is being typed in. While typing I'm feeding back to Typeahead whatever was typed in that particular entry (with the exception of resetting it to empty string initially.)

While I do have some extra state to deal with when it comes to Typeahead I'm only really feeding the value from `onInputChange` back (wrapped as an object as appropraite) as in the test case.

So, currently I'm trying to use Typeahead as controlled component (may need to revisit that) and finding a bit of a gap when it comes to arbitrary text not present in the options list. When making that new option on the fly should I add it to the options list? Should Typeahead deal with `selected` value being something that is _not_ in `options`?

**Comment by ericgio** [comment]
Can you explain the need for two typeaheads? Why not filter on both email and name in a single typeahead?

**Comment by tko** [comment]
One entry to get the name, the other one to get the email. Both filter on both email and name of existing users, and both entries are needed to get name+email for a new person.

**Comment by ericgio** [comment]
@tko: I've fixed some bugs in v2.0.0-alpha.1 that were related to the issues you were having, but I'm not sure if it ultimately solves your issue. If you want to post a full repro case, I can have a look.

However, what I think you're asking for is the ability to set the input's `value`. This is currently still not possible, and I'm not sure it will be.

**Comment by ericgio** [comment]
I believe the actual bugs have been addressed here and this is either a feature request or just not using the component correctly. If you want to post a full repro case (jsfiddle, test repo) I can have a look, but will consider this issue closed otherwise.
